### DISCUSSION TITLE
Nested matrix breaks

### DISCUSSION THREAD

**Discussion by ndrean** [KaTeX/KaTeX#3782] [Q&A]
Hi there,
Did someone succeed in writing nested matrices? Something like this. For me, it breaks, and puts random "```" signs. `pmatrix` and `bmatrix` behave the same.

```
\begin{bmatrix}
   \begin{bmatrix}
      \begin{bmatrix}
      \end{bmatrix}
      \begin{bmatrix}
      \end{bmatrix}
   \end{bmatrix}
   \begin{bmatrix}
      \begin{bmatrix}
      \end{bmatrix}
      \begin{bmatrix}
      \end{bmatrix}
   \end{bmatrix}
\end{bmatrix}
```

**Comment by edemaine** [comment]
Looks fine on the demo site:

![image](https://user-images.githubusercontent.com/2218736/217882964-0ec9ec9b-e211-402f-9970-9d9117dfc9c4.png)

Perhaps an issue with how you're calling KaTeX? Can you show a screenshot and ideally the source?

**Comment by ndrean** [comment]
Yes I tried this as in the screen above. I am probably wrong to ask this here but to give some context, I am using [Livebook](https://livebook.dev/), an Elixir app that can render markdown and  LAtex with Katex.

The exact same code as above renders this:
<img width="272" alt="Screenshot 2023-02-09 at 18 49 44" src="https://user-images.githubusercontent.com/6793008/217897160-1a0f4126-774d-4b28-9329-204c439e8411.png">

And the code is transformed as below. It adds 3 commas "```"  after the second  nesting so I was puzzled. Wasn't sure of the origin

The same latex code becomes:

\begin{bmatrix}
   \begin{bmatrix}

    "```"
  \begin{bmatrix}
  \end{bmatrix}
  \begin{bmatrix}
  \end{bmatrix}
  "```"

\end{bmatrix}
   \begin{bmatrix}

 "```"
 

  \begin{bmatrix}
  \end{bmatrix}
  \begin{bmatrix}
  \end{bmatrix}

"```"

\end{bmatrix}
\end{bmatrix}

**Reply by edemaine** [reply]
It's presumably a bug (or limitation) of Livebook, so maybe worth reporting there.  I assume you're wrapping the code in `$$`?  It looks like Livebook uses [rehype-katex](https://www.npmjs.com/package/rehype-katex) so it might also be an issue with that...

**Reply by ndrean** [reply]
Yes indeed, I removed `$$` for readability. Many thanks for your help!
